src/mmw/js/src/data_catalog/models.js

```diff
@@ -3,38 +3,122 @@
 var $ = require('jquery'),
     _ = require('lodash'),
     Backbone = require('../../shim/backbone'),
+    moment = require('moment'),
     settings = require('../core/settings');
 
 var REQUEST_TIMED_OUT_CODE = 408;
 var DESCRIPTION_MAX_LENGTH = 100;
 var PAGE_SIZE = settings.get('data_catalog_page_size');
 
-var SearchOption = Backbone.Model.extend({
+
+var FilterModel = Backbone.Model.extend({
     defaults: {
         id: '',
-        active: false,
+        type: '',
+        isValid: true
+    },
+
+    reset: function() {
+        this.clear({ silent: true }).set(this.defaults());
+    },
+
+    validate: function() {
+        _.noop();
+    },
+
+    isActive: function() {
+        window.console.error("Use of unimplemented function",
+                             "FilterModel.isActive");
+        return false;
+    }
+});
+
+var SearchOption = FilterModel.extend({
+    defaults: function() {
```

Comment:
Since `FilterModel.prototype.defaults` is defined at "compile time" this doesn't need to be a dynamic function. Instead we can simply do:

```js
defaults: _.defaults({
    active: false,
}, FilterModel.prototype.defaults),
```

For example, see https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/js/src/core/modals/views.js#L90-L93

Reply:
Awesome. Thanks!